### PR TITLE
Sort nutrients alphabetically in product editing

### LINKED ISSUE #3086: Sort the nutrients list by alphabetical order in product editing
Sort the nutrients list by alphabetical order so it is easier to search for a nutrient.

Orignal comment

> Cuando se agrega un nuevo nutriente a la lista de nutrientes del producto. La lista que sale no esta ordenada alfabeticamente. Es un infierno buscar el nuevo nutriente.

9829a56d-8b66-4ab7-aa57-385fe5e7ee20

### PR DESCRIPTION
**Pull request by raychenon**
Fixes #3086

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java [hunk 1]
```diff
@@ -754,17 +755,19 @@ private boolean isDataPer100() {
 
     @OnClick(R.id.btn_add_a_nutrient)
     void addNutrient() {
+        String[] nutrients = getResources().getStringArray(R.array.nutrients_array);
+        Arrays.sort(nutrients);
```
Comment (raychenon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java:759]: This sort processes only unicode characters.

```
["Zorro", "Almidón", "Ácido esteárico"]  // before sort
["Almidón", "Zorro", "Ácido esteárico"]  // after sort
```
`Á` comes after  `Z`.

Need to write a custom sort for later.

Comment (raychenon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java]: Now the sorting method ignore the accents.
```
["Ácido esteárico", "Almidón", "Zorro"]  // after sort
```

See the tests in `StringComparatorTest`.
I don't know if `StringComparator` is in the right package `utils`.

**Comment by teolemon** [comment]
I was looking at Collator. Good idea or overkill ?
https://stackoverflow.com/questions/7633552/is-there-any-way-to-sort-strings-in-all-languages

**Comment by raychenon** [comment]
Actually with `Collator` , it is easier. `Comparator` was overkill.

The order changes
```
{"Molybdène", "manganèse", "molybdene", "Manganèse"}   // before sort
{"Manganèse", "Molybdène", "manganèse", "molybdene"}   // sort with Comparator
{"manganèse", "Manganèse", "molybdene", "Molybdène"}   // sort with Collator
```

http://tutorials.jenkov.com/java-internationalization/collator.html

I can push the changes tomorrow.

**Comment by VaiTon** [comment]
Superseeded by #3089.

**Pull request closed without merging**